### PR TITLE
fix: Argument for @NotNull parameter 'file' of com/intellij/openapi/vfs/VfsUtilCore.virtualToIoFile must not be null

### LINKED ISSUE #1228: Argument for @NotNull parameter 'file' of com/intellij/openapi/vfs/VfsUtilCore.virtualToIoFile must not be null
I regularly see those errors:

```
java.lang.IllegalArgumentException: Argument for @NotNull parameter 'file' of com/intellij/openapi/vfs/VfsUtilCore.virtualToIoFile must not be null
	at com.intellij.openapi.vfs.VfsUtilCore.$$$reportNull$$$0(VfsUtilCore.java)
	at com.intellij.openapi.vfs.VfsUtilCore.virtualToIoFile(VfsUtilCore.java)
	at com.redhat.devtools.intellij.lsp4ij.LSPIJUtils.toUri(LSPIJUtils.java:124)
	at com.redhat.devtools.intellij.lsp4ij.operations.codelens.LSPCodelensInlayProvider$1.collect(LSPCodelensInlayProvider.java:79)
	at com.intellij.codeInsight.hints.CollectorWithSettings.collectHints(InlayHintsUtils.kt:91)
	at com.intellij.codeInsight.hints.InlayHintsPass.doCollectInformation$lambda$0(InlayHintsPass.kt:46)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:142)
```

### PR DESCRIPTION
**Pull request by angelozerr**
fix: Argument for @NotNull parameter 'file' of com/intellij/openapi/vfs/VfsUtilCore.virtualToIoFile must not be null

Fixes #1228

### CODE REVIEW

File: src/main/java/com/redhat/devtools/intellij/lsp4ij/operations/navigation/LSPGotoDeclarationHandler.java [hunk 1]
```diff
@@ -51,39 +51,40 @@ public class LSPGotoDeclarationHandler implements GotoDeclarationHandler {
...
-                LOGGER.warn(e.getLocalizedMessage(), e);
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
+        Document document = editor.getDocument();
+        VirtualFile file = LSPIJUtils.getFile(document);
+        if (file == null) {
+            return PsiElement.EMPTY_ARRAY;
+        }
+        URI uri = LSPIJUtils.toUri(file);
+        DefinitionParams params = new DefinitionParams(LSPIJUtils.toTextDocumentIdentifier(uri), LSPIJUtils.toPosition(offset, editor.getDocument()));
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/lsp4ij/operations/navigation/LSPGotoDeclarationHandler.java:60]: LSPIJUtils.toPosition(offset, document)

**Comment by fbricon** [comment]
LSPCompletionContributor.initiateLanguageServers also need to check for non-null file

**Comment by fbricon** [comment]
https://github.com/redhat-developer/intellij-quarkus/blob/b3252cc7148d62b814c4de4471e56c93126e4a4d/src/main/java/com/redhat/devtools/intellij/lsp4ij/operations/highlight/LSPHighlightUsagesHandlerFactory.java#L72 should do

```java
.getLanguageServers(file, capabilities -> LSPIJUtils.hasCapability(capabilities.getDocumentHighlightProvider()))
```

**Review by fbricon — CHANGES_REQUESTED** [review]

File: src/main/java/com/redhat/devtools/intellij/lsp4ij/operations/navigation/LSPGotoDeclarationHandler.java (hunk 1 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/lsp4ij/operations/navigation/LSPGotoDeclarationHandler.java]: fixed

**Pull request merged**